### DISCUSSION TITLE
Focus field removal on form submit

### DISCUSSION THREAD

**Discussion by Maarten-Dekker** [livewire/livewire#7324] [3. Feature ideas]
When you hit enter from within an input element, the input element keeps focus (unless you redirect, of course). This is often not the desired result because, after sending the form, the same action rarely needs to be done again. When having multiple input elements and multiple input errors, the focus stays most likely on the last input element, so it's not making things easier in that case.

I would propose a `$this->blur();`* action to remove the focus of all (specified) elements, or to implement this as the default when a form is being submitted. Right now, you can remove focus by clearing all fields (by providing `$this->reset();`), but this will prevent the user from seeing the submitted values.

If there is already an easy solution to this, please point me to it :).

*The blur() naming is taken from the JS world (`document.getElementById("focus").blur();`)

**Comment by WilliamPinto-Olmos** [comment]
hey @Maarten-Dekker i think you can do `$this->js("document.activeElement.blur()");` from your component.

also if you want a more likeable syntax, you can create a Macro in your `AppServiceProvider.php` boot method

```php
use Livewire\Component;

Component::macro('blur', function ($message) {
   $this->js("document.activeElement.blur()");
});
```

and call it as 

```php
$this->blur();
```

from any of your livewire components. I have not tested it tho, so lmk if it works.

**Reply by Maarten-Dekker** [reply]
I didn't know there existed something like activeElement.
Thanks for your reply!
